testing.rb

```diff
@@ -0,0 +1,311 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/BlockNesting,Metrics/MethodLength,Metrics/ModuleLength,Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity,Metrics/AbcSize
+
+# module Enumerable
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+    arr = to_a if self.class == Hash
+    n = 0
+    while n < arr.length
+      yield(arr[n])
+      n += 1
+    end
+    arr
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+    arr = to_a if self.class == Hash
+    n = 0
+    while n < arr.length
+      yield(arr[n], n)
+      n += 1
+    end
+    arr
+  end
+
+  def my_none?(input = nil)
+    return true if !block_given? && empty?
+
+    if !block_given? && input == Float
+      my_each do |item|
+        return false if item.class == input
+      end
+    elsif !block_given? && input == String
+      my_each do |item|
+        return false if item.class == input
+      end
+      true
+    elsif !block_given? && input == Numeric
+      my_each do |item|
+        return false if item.class == input
+      end
+      true
+    elsif !block_given? && input.class == Regexp
+      my_each  do |item|
+        char = item.split('')
+        char.my_each do |n|
+          return false if n.match?(input)
+        end
+      end
+      true
+    elsif !block_given? && input.nil?
+      my_each do |item|
+        return false if item == true
+      end
+      true
+    elsif !block_given? && !input.nil?
+      my_each do |item|
+        return false if item.to_s == input.to_s
+      end
+      true
+    elsif block_given?
+      my_each do |item|
+        return false if item == true
+        return false if yield(item)
+      end
+      true
+    end
+  end
+
+  def my_count(*args)
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+
+    if !block_given? && args[0].nil?
+      counter = 0
+      counter += 1 while counter < arr.length
+      counter
+    elsif !block_given? && !args[0].nil?
+      count = 0
+      section = 0
+      while count < length
+        section += 1 if arr[count] == args[0]
+        count += 1
+      end
+      section
+    elsif block_given? && args[0].nil?
+      num = 0
+      ematch = 0
+      while num < length
+        ematch += 1 if yield(arr[num])
+        num += 1
+      end
+      ematch
+    end
+  end
+
+  def my_map
+    return to_enum(:my_map) unless block_given?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+    counter = 0
+    tot = []
+    while counter < arr.length
+      runn = yield(arr[counter])
+      tot.push(runn)
+      counter += 1
+    end
+    tot
+  end
```

Comment:
Your #my_map does not handle cases when a `proc` is given:
Your method should look like this: `def my_map(proc = nil)`
Within the method add the logic that handles a proc if given. [See docs](https://ruby-doc.org/core-2.6/Proc.html) about proc